enterprise/internal/batches/store/author/author_test.go

```diff
@@ -0,0 +1,92 @@
+package author
+
+import (
+	"context"
+	"testing"
+
+	"github.com/sourcegraph/log/logtest"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbtest"
+)
+
+func TestGetChangesetAuthorForUser(t *testing.T) {
+
+	logger := logtest.Scoped(t)
+	ctx := context.Background()
+	db := database.NewDB(logger, dbtest.NewDB(logger, t))
+
+	userStore := db.Users()
+
+	t.Run("User ID doesnt exist", func(t *testing.T) {
+		author, err := GetChangesetAuthorForUser(ctx, userStore, 0)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if author != nil {
+			t.Fatalf("got non-nil author email when author doesnt exist: %v", author)
+		}
+	})
+
+	t.Run("User exists but doesn't have an email", func(t *testing.T) {
+
+		user, err := userStore.Create(ctx, database.NewUser{
+			Username: "mary",
+		})
+		if err != nil {
+			return
```

Comment:
We should probably fail the test if we got an error creating the user, otherwise the test is still going to look liked it succeeded even though it isn't actually testing the thing we want it to:

```suggestion
			t.Fatalf("failed to create test user: %v", err)
```